crates/oxc_linter/src/rules/eslint/sort_keys.rs

```diff
@@ -0,0 +1,1089 @@
+
+            for (i, prop) in dec.properties.iter().enumerate() {
+                if let ObjectPropertyKind::SpreadProperty(_) = prop {
+                    property_groups.push(vec!["<ellipsis_group>".into()]);
+                    property_groups.push(vec![]);
+                    continue;
+                }
+                let key = match prop.prop_name() {
+                    Some((name, _)) => name,
+                    // FIXME: prop_name is currently unset for computed properties, short hand properties, and function declarations.
+                    None => extract_property_key(prop.span().source_text(source_text)),
+                };
+
+                if i != dec.properties.len() - 1 && self.allow_line_separated_groups {
+                    let text_between = extract_text_between_spans(
+                        source_text,
+                        prop.span(),
+                        dec.properties[i + 1].span(),
+                    );
+                    if text_between.contains("\n\n") {
+                        property_groups.last_mut().unwrap().push(key.into());
+                        property_groups.push(vec!["<linebreak_group>".into()]);
+                        property_groups.push(vec![]);
+                        continue;
+                    }
+                }
+                property_groups.last_mut().unwrap().push(key.into());
+            }
+
+            if !self.case_sensitive {
+                for group in &mut property_groups {
+                    *group = group.iter().map(|s| s.to_lowercase()).collect::<Vec<String>>();
+                }
+            }
+
+            let mut sorted_property_groups = property_groups.clone();
+            for group in &mut sorted_property_groups {
+                if self.natural {
+                    natural_sort(group);
+                } else {
+                    alphanumeric_sort(group);
+                }
+
+                if self.sort_order == SortOrder::Desc {
+                    group.reverse();
+                }
+            }
+
+            let is_sorted =
+                all(property_groups.iter().zip(&sorted_property_groups), |(a, b)| a == b);
+
+            if !is_sorted {
+                ctx.diagnostic(sort_properties_diagnostic(node.span()));
+            }
+        }
+    }
+}
+
+fn alphanumeric_cmp(a: &str, b: &str) -> Ordering {
+    /* regex key special case */
+    if a.starts_with('/') && a.ends_with('/') {
+        if b.starts_with('/') && b.ends_with('/') {
+            return a.cmp(b);
+        }
+        return Ordering::Greater;
+    }
+
+    if b.starts_with('/') && b.ends_with('/') {
+        return Ordering::Less;
+    }
+
+    /* empty keys special case */
+    if a.is_empty() && b.starts_with('[') || b.is_empty() && a.starts_with('[') {
+        return Ordering::Equal;
+    }
+
+    let len = a.len().min(b.len());
+    let a_chars: Vec<char> = a.chars().take(len).collect();
```

Comment:
Can you zip these two iterations together without collecting?

Reply:
done